pkg/formatted/color.go

```diff
@@ -16,19 +16,78 @@ package formatted
 
 import (
 	"io"
+	"sync"
+	"sync/atomic"
 
 	"github.com/fatih/color"
 )
 
+var (
+	// Red is avoided as keeping it for errors
+	palette = []color.Attribute{
+		color.FgHiGreen,
+		color.FgHiYellow,
+		color.FgHiBlue,
+		color.FgHiMagenta,
+		color.FgHiCyan,
+	}
+)
+
+type atomicCounter struct {
+	value     uint32
+	threshold int
+}
+
+func (c *atomicCounter) next() int {
+	v := atomic.AddUint32(&c.value, 1)
+	next := int(v-1) % c.threshold
+	atomic.CompareAndSwapUint32(&c.value, uint32(c.threshold), 0)
+	return next
+
+}
+
+type rainbow struct {
+	cache   sync.Map
+	counter atomicCounter
+}
+
+func newRainbow() *rainbow {
+	return &rainbow{
+		counter: atomicCounter{threshold: len(palette)},
+	}
+}
+
+func (r *rainbow) get(x string) color.Attribute {
+	if value, ok := r.cache.Load(x); ok {
+		return value.(color.Attribute)
+	}
+
+	clr := palette[r.counter.next()]
+	r.cache.Store(x, clr)
+	return clr
+}
+
+// Fprintf formats according to a format specifier and writes to w.
+// the first argument is a label to keep the same colour on.
+func (r *rainbow) Fprintf(label string, w io.Writer, format string, args ...interface{}) {
```

Comment:
imho, should follow the same signature/convention as Fprintf which is to accept `io.Writer` as the first argument.

Reply:
I think this change can be a follow up patch. 
hence lgtm.

Reply:
it would never be the same as Fprintf since it need to be 'labelled', i can rename it to LabelFrprintf if that cause for confusion,